src/shaders/fill_extrusion.vertex.glsl

```diff
@@ -117,33 +117,44 @@ void main() {
     v_depth = gl_Position.w;
 #endif
 
+    float NdotL = 0.0;
+    float colorvalue = 0.0;
+#ifdef LIGHTING_3D_MODE
+    const float ext = 0.70710678118; // acos(pi/4)
+    NdotL = (clamp(dot(normal, u_lighting_directional_dir), -ext, 1.0) + ext) / (1.0 + ext);
```

Comment:
Would it be possible to hide these inside the prelude file too?

Reply:
I think this deserves a comment on why we modify NdotL like (from @mpulkki-mapbox PR on the Poc): 
```
Use slightly modified dot product for lambertian diffuse shading. This increase the range of NdotL to cover surfaces facing up to 45 degrees away from the light source. This allows us to trade some realism for performance/usability as a single light source is enough to shade the scene.
```